R/particle_filter.R

```diff
@@ -247,11 +252,18 @@ particle_filter <- R6::R6Class(
         stop("nested data not supported for continuous models")
       }
 
-      if (!is_continuous && !is.null(stochastic_schedule)) {
-        stop(paste("'stochastic_schedule' provided but 'model' does not",
-         "support this"))
+      if (!is_continuous) {
+        if (!is.null(stochastic_schedule)) {
+          stop(paste("'stochastic_schedule' provided but 'model' does not",
+           "support this"))
+        }
+        if (!is.null(control)) {
+          stop(paste("'control' provided but 'model' does not",
+                     "support this"))
+        }
       } else {
         private$stochastic_schedule <- stochastic_schedule
+        private$control <- control
```

Comment:
Add

```
assert_is(control, "mode_control")
```

to validate that the control is ok here, and not when we initialise the mode model later.

(renaming control -> ode_control if you agree with the above)